core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/predicateJoin.kt

```diff
@@ -0,0 +1,50 @@
+package org.jetbrains.kotlinx.dataframe.api
+
+import org.jetbrains.kotlinx.dataframe.DataFrame
+import org.jetbrains.kotlinx.dataframe.DataRow
+import org.jetbrains.kotlinx.dataframe.Selector
+import org.jetbrains.kotlinx.dataframe.impl.api.predicateJoinImpl
+
+public interface JoinedDataRow<out A, out B> : DataRow<A> {
+    public val right: DataRow<B>
+}
+
+public typealias JoinExpression<A, B> = Selector<JoinedDataRow<A, B>, Boolean>
+
+public fun <A, B> DataFrame<A>.predicateJoin(
+    right: DataFrame<B>,
+    type: JoinType = JoinType.Inner,
+    joinExpression: JoinExpression<A, B>
+): DataFrame<A> {
+    return predicateJoinImpl(right, type, addNewColumns = type.addNewColumns, joinExpression)
```

Comment:
same as in join, I'd make it an expression function and provide named arguments for all